diff --git a/profiles/dkan/modules/dkan/dkan_dataset/dkan_dataset.theme.inc b/profiles/dkan/modules/dkan/dkan_dataset/dkan_dataset.theme.inc
index e435e49..23bdda6 100755
--- a/profiles/dkan/modules/dkan/dkan_dataset/dkan_dataset.theme.inc
+++ b/profiles/dkan/modules/dkan/dkan_dataset/dkan_dataset.theme.inc
@@ -120,7 +120,7 @@ function theme_dkan_dataset_resource_view($vars) {
     }
     if ($file_count > 1) {
       $download_all = '<div property="dcat:Distribution"><span class="links">';
-      $download_all .= l('Download All<i class="fa fa-download"><span> Download</span></i>', 'node/' . $vars['node']->nid . '/dataset/download', array(
+      $download_all .= l('Скачать Всё<i class="fa fa-download"><span> Download</span></i>', 'node/' . $vars['node']->nid . '/dataset/download', array(
         'html' => TRUE,
         'attributes' => array(
           'class' => array('btn', 'btn-primary')
