diff --git a/search_api_db.install b/search_api_db.install
index ae604e3..929bc19 100644
--- a/search_api_db.install
+++ b/search_api_db.install
@@ -194,14 +194,26 @@ function search_api_db_update_7104() {
         $connection->query("ALTER TABLE {{$text_table}} CONVERT TO CHARACTER SET 'utf8' COLLATE 'utf8_bin'");
       }
       // Migrate data.
+      // Some tables may already be reusing (in case of not updating or features
+      // so track if used more than once and only delete on 0 usage.
+      $usage = array();
       foreach ($options['indexes'][$server->machine_name] as $name => $field) {
-        if (search_api_is_text_type($field['type'])) {
-          $query = $connection->select($field['table'], 't')
-            ->fields('t', array('item_id', 'word', 'score'));
-          $query->addExpression(':field_name', 'field_name', array(':field_name' => strlen($name) > 255 ? md5($name) : $name));
-          $connection->insert($text_table)->from($query)->execute();
-
-          $connection->schema()->dropTable($field['table']);
+        $usage[$field['table']] = isset($usage[$field['table']]) ? $usage[$field['table']] + 1 : 1;
+      }
+      foreach ($options['indexes'][$server->machine_name] as $name => $field) {
+        if (search_api_is_text_type($field['type']) && ($field['table'] != $text_table)) {
+          // The table may not exist (sometimes in case of features).
+          if ($connection->schema()->tableExists($field['table'])) {
+            $query = $connection->select($field['table'], 't')
+              ->fields('t', array('item_id', 'word', 'score'));
+            $query->addExpression(':field_name', 'field_name', array(':field_name' => strlen($name) > 255 ? md5($name) : $name));
+            $connection->insert($text_table)->from($query)->execute();
+  
+            $usage[$field['table']]--;
+            if (empty($usage[$field['table']])) {
+              $connection->schema()->dropTable($field['table']);
+            }
+          }
           $server_options[$server->server]['indexes'][$server->machine_name][$name]['table'] = $text_table;
         }
       }
