diff --git a/l10n_update.install b/l10n_update.install
index 5ea0845..68347f0 100644
--- a/l10n_update.install
+++ b/l10n_update.install
@@ -15,7 +15,7 @@ function l10n_update_schema() {
       'name' => array(
         'description' => 'A unique short name to identify the project.',
         'type' => 'varchar',
-        'length' => '50',
+        'length' => '255',
         'not null' => TRUE,
       ),
       'project_type' => array(
@@ -367,3 +367,11 @@ function l10n_update_update_7203() {
     db_drop_field('l10n_update_project', 'l10n_server');
   }
 }
+
+/**
+ * Increase the length of the name column.
+ */
+function l10n_update_update_7204() {
+  $schema = l10n_update_schema();
+  db_change_field('l10n_update_project', 'name', 'name', $schema['l10n_update_project']['fields']['name']);
+}
